pkg/checker/checker.go

```diff
@@ -2,49 +2,113 @@ package checker
 
 import "fmt"
 
-// Checker is the type of checking functions
-type Checker func() error
+// ResultOK is the string to appear in CheckResult.Result if the check was successful
+const ResultOK = "OK"
 
-// Remediator is the type of remediation functions
-type Remediator func() error
+//ResultError is the string to appear in CheckResult.Result if the check failed
+const ResultError = "ERROR"
 
-// Check is a type that implements a netboot check
+// CheckFun is the type of checking functions
+type CheckFun func(CheckArgs) error
+
+// CheckArgs is the argument map to be passed to CheckFuns
+type CheckArgs map[string]string
+
+// A Check describes a call to a CheckFun with possible remediations
 type Check struct {
-	Name        string
-	Run         Checker
-	Remediate   Remediator
-	StopOnError bool
+	Description string `json:"description"`
+
+	CheckFunName string    `json:"checkFun"`
+	CheckFunArgs CheckArgs `json:"checkArgs"`
+
+	Remediations  []Check `json:"remediations"`
+	StopOnFailure bool    `json:"stopOnFailure"`
+}
+
+// A CheckResult describes the result of Running a Check
+type CheckResult struct {
+	Description string `json:"description"`
+
+	CheckFunName string    `json:"checkFun"`
```

Comment:
I guess we should call this `CheckerFuncName` ("func" instead of "fun") for consistency and clarity